mantidproject/mantid
# IndirectReplaceFitResult will replace data based on the parameter name in the y axis

**Description of work.**
This PR changes the IndirectReplaceFitResult so that it will replace data in one workspace using data in another workspace using the parameter names stored in each row.

This allows there to be a differing number of histograms for each of the input workspaces.
It also means data will be replaced correctly if the two workspaces contain the same parameters but in a different order. Previously it assumed the parameters were displayed in the same order.

**To test:**
1. `Interfaces`->`Indirect`->`Data Analysis`->`ConvFit` tab
2. Load the data below
3. Select `One Lorentzian`
4. Click `Run` and then `Fit Single Spectrum`
5. Click `Edit Results` and choose an output name
6. Click `Replace Fit Result`. There should be no pop-up box saying the number of histograms are different.

**Data Files**
[irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/3059890/irs26176_graphite002_red.zip)
[irs26173_graphite002_res.zip](https://github.com/mantidproject/mantid/files/3059892/irs26173_graphite002_res.zip)

Fixes #25490

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
